### DISCUSSION TITLE
How to get the media_url of a tweet when using get_all_tweets function?

### DISCUSSION THREAD

**Discussion by keevinhoo** [cjbarrie/academictwitteR#293] [Q&A]
Hello guys, 

at first, kudos to such a great package. 

I am wondering whether it is possible to retrieve the media_url of an image that has been posted within a tweet in Twitter. Right now, I am only able to get the attachments.media_keys variable. 
Maybe there is a way to link that key to the url... 

I would be grateful for any advice/help.

Thank you all in advance!

**Comment by chainsawriot** [comment]
@keevinhoo see #255 but also the problems.

**Reply by keevinhoo** [reply]
Thank you @chainsawriot. I have had a look at it but when incorporating the very same approach as you I am getting a warning message: Unknown or uninitialised column: `images` and a tibble: 0 x 0 most of the times. But sometimes I am also getting the urls with varying output. This seems pretty random here. 

I am trying to loop over a list of users in order to get their tweets. The only thing I am lacking, is the tweet url, respectively the tweet jpg in my current code. I tried to use the code snipped you provided, but unfortunately that did not work out. If it worked out, I would add these tweet urls as a new column to my existing data frame.  The rest of the code works perfectly fine for me. I would higly appreciate any advice/help. Thank you in advance guys!

In order to get you a better understanding of my problem, I am using the following code: 

require(academictwitteR)
library("writexl")
library(jsonlite)
library(data.table)
library(purrr)
library(dplyr)
library(tidyverse)

set_bearer()
get_bearer()
setwd("/Users/Kevin/example")
getwd()
handles <- read.csv("example.csv")
df <- data.frame(handles)
x <- c(t(df))

for (i in seq_along(x)) {
  tweets <- get_all_tweets(
    users = x[i],
    start_tweets = "2020-05-26T00:00:00Z",
    end_tweets = "2020-05-29T00:00:00Z",
    is_retweet=FALSE,
    is_reply=FALSE,
    data_path = "data",
    verbose = TRUE,
    bind_tweets = FALSE,
    export_query = TRUE,
    n=Inf,
    context_annotations=TRUE
  )
}

data<-bind_tweets(data_path="data/",user = FALSE)
users<-bind_tweets(data_path="data/",user = TRUE)

purrr::keep(data$entities$urls, ~"images" %in% colnames(.)) %>% purrr::map_dfr(~.) -> images
images$images %>% purrr::discard(is.null) %>% purrr::map_dfr(~.)

data <- as.data.table(data)
users <- as.data.table(users)

names(users)[names(users)=="id"] <- "author_id"

data <- merge(data,users,by="author_id")

saveRDS(data,"Tweets.rds")

write.csv(data, file = "twitter_tweets.csv", row.names=FALSE)
